### PR TITLE
Add one-time-password extension

### PR DESCRIPTION
**Pull request by lacherogwu**
## Description

# One Time Password

Generate time-based one-time passwords

The Raycast One Time Password extension provides a convenient way to manage your One Time Passwords. With the ability to scan QR codes or manually enter secrets, the extension generates and displays Time-based OTPs.

Users can easily name their keys, add new ones, remove them, and view them in an intuitive and easy-to-use design.

This extension simplifies the process of managing OTPs and makes it easy to access them when needed.

## Supports Google Authenticator Migration

To migrate Google Authenticator, follow these steps:

1. Export your accounts from the “Authenticator” app by Google
2. Scan the QR code

## Credits

- Inspiration from [jaden/totp-generator](https://github.com/jaden/totp-generator)
- [Export Google Authenticator secret OTP-keys](https://gist.github.com/mapster/4b8b9f8f6b92cc1ca58ae5506e0508f7)

## Screencast

one-time-password-1
one-time-password-2
one-time-password-3
one-time-password-4
one-time-password-5
one-time-password-6

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by lacherogwu** [comment]
@Aayush9029 Did my best here 🤓 hope you will like the changes

btw why lint complains?
CleanShot 2023-05-05 at 02 30 44

I've used the `Raycast Window Capture`

**Comment by Aayush9029** [comment]
> Did my best here 🤓 hope you will like the changes

love it

**Comment by lacherogwu** [comment]
> 

Yeah I am using external monitor, Okay let me re-take them

**Comment by lacherogwu** [comment]
I added support to import accounts from Google Authenticator, but encountered an issue while trying to save the `﻿migrationPayload` (in `google-authenticator.ts`) string in a dedicated file called `﻿migration.proto`. My plan was to load the file using ﻿`protofuf.load()` with a path parameter. However, when I provided the path parameter, the file couldn’t be found. I realized that the extension was saved in the build code, which doesn’t have the ﻿src directory, only the ﻿assets directory. 
The path to the file was `﻿/Users/asaf/.config/raycast/extensions/one-time-password`. I’m unsure if the ﻿assets directory is the appropriate place for other files.

File: extensions/one-time-password/src/one-time-password.tsx [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    setTimer(TOKEN_TIME - (getCurrentSeconds() % TOKEN_TIME));
+  }
+
+  function getProgressColor() {
+    if (timer > 5) {
+      return '#61b972'; // green
+    } else {
+      return '#d26060'; // red
+    }
+  }
```
Comment (pernielsentikaer) [inline extensions/one-time-password/src/one-time-password.tsx:117]: Could be better to rely on Color.Green / Color.Red from Raycast?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by pernielsentikaer** [comment]
Hi @lacherogwu 👋 

You should use [supportPath](https://developers.raycast.com/api-reference/environment#properties) for your files - could you try that 🙂

**Comment by lacherogwu** [comment]
> Hi @lacherogwu 👋
> 
> You should use [supportPath](https://developers.raycast.com/api-reference/environment#properties) for your files - could you try that 🙂

I think that doesn't help.
I want the file to be stored in the source code somewhere and then to read it in runtime

**Comment by pernielsentikaer** [comment]
And you already tried adding it to the `assetsPath` folder? I'm not sure what you're trying to achieve, is there no way to load from a node_module?

**Comment by lacherogwu** [comment]
No I haven’t try, I was just asking if assets will be the right place for
this file

**Comment by lacherogwu** [comment]
No I haven’t try, I was just asking if assets will be the right place for this file

**Comment by pernielsentikaer** [comment]
We have other extension that have some files in the `assets` folder

**Comment by lacherogwu** [comment]
Okay so I guess it's the right place, let me add it there

**Comment by lacherogwu** [comment]
@pernielsentikaer You can now review all at once

File: extensions/one-time-password/src/store.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    account.secret = secret || '';
+    account.issuer = issuer || '';
+  }
+
+  accounts.push({ ...account, id: generateAccountId(account) });
+  await save(accounts);
+}
+
+export async function removeAccount(id: string) {
+  const accounts = await getAccounts();
```
Comment (pernielsentikaer) [inline extensions/one-time-password/src/store.ts:51]: I would add a [confirmAlert](https://developers.raycast.com/api-reference/feedback/alert#confirmalert) here 🙂

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**